utils/components-utils/src/main/scala/pl/touk/nussknacker/engine/util/sinkvalue/SinkValueData.scala

```diff
@@ -31,13 +38,40 @@ object SinkValueData {
 
   sealed trait SinkValueParameter {
     def toParameters: List[Parameter] = this match {
-      case SinkSingleValueParameter(value) => value :: Nil
+      case SinkSingleValueParameter(value, _) => value :: Nil
       case SinkRecordParameter(fields) => fields.values.toList.flatMap(_.toParameters)
     }
+    def validateParams(resultType: List[(String, BaseDefinedParameter)], prefix: List[String])(implicit nodeId: NodeId): ValidatedNel[ProcessCompilationError, Unit]
   }
 
-  case class SinkSingleValueParameter(value: Parameter) extends SinkValueParameter
+  case class SinkSingleValueParameter(value: Parameter, validator: SchemaOutputValidator) extends SinkValueParameter {
+    override def validateParams(resultTypes: List[(String, BaseDefinedParameter)], prefix: List[String])(implicit nodeId: NodeId): ValidatedNel[ProcessCompilationError, Unit] = resultTypes match {
+      case (fieldName, resultType) :: Nil =>
+        val converter = new OutputValidatorErrorsConverter(fieldName)
+        validator
+          .validateTypingResultAgainstSchema(resultType.returnType)
+          .leftMap(converter.convertValidationErrors)
+          .leftMap(NonEmptyList.one)
+      case _ => Validated.invalidNel(CannotCreateObjectError("Unexpected parameter list", nodeId.id))
+    }
+  }
 
-  case class SinkRecordParameter(fields: ListMap[FieldName, SinkValueParameter]) extends SinkValueParameter
-}
+  case class SinkRecordParameter(fields: ListMap[FieldName, SinkValueParameter]) extends SinkValueParameter {
+    override def validateParams(actualResultTypes: List[(String, BaseDefinedParameter)], path: List[String])(implicit nodeId: NodeId): ValidatedNel[ProcessCompilationError, Unit] = {
+      fields.map { case (fieldName, sinkValueParameter) =>
+        val fieldPath = path :+ fieldName
+        val fieldsMatchingPath = actualResultTypes.filter { case (name, _) => name.startsWith(fieldPath.mkString(".")) }
```

Comment:
IMO that `startWith` approach won't work correctly in many cases. Consider situation when you have schema like:
```
foo: String
fooBar: Integer
```
You can't be sure if `foo` will be validated against `fooBar` schema. WDYT about exact checking parameter name. I did such a change in #4013 and it looks like all tests passed. Also thanks to this approach, you don't need to rebuild "path" (one parameter less in `validateParams` method).

Reply:
thansk, merged